version 0.8

EvgeniyKuch opened the pull request:
Приложение, хранящее информацию о книгах в библиотеке

Использованы возможности Spring Data для подключения к нереляционным базам данных.
Web-интерфейс выполнен с применением Thymeleaf.

Хранение данных выполняется в Mongo DB.
Структуру и начальные данные создаёт Mongock.

Commit:
version 0.8

Приложение, хранящее информацию о книгах в библиотеке

Использованы возможности Spring Data для подключения к нереляционным базам данных.
Web-интерфейс выполнен на Thymeleaf.

Хранение данных выполняется в Mongo DB.
Структуру и начальные данные создаёт Mongock.

Commit:
version 0.9

В CommentController в handler’е «/comment/add-for-book/{id}» метод getBookWithCommentsById() заменен на getBookById(), потому что для рендеринга страницы добавления комментариев не нужны уже существующие комментарии.

kataus commented on library-spring-data-mongo-mvc-view/src/main/java/ru/otus/library/controller/CommentController.java:
```diff
@@ -0,0 +1,47 @@
+import ru.otus.library.service.CommentService;
+
+@Controller
+@AllArgsConstructor
```
Порядок аннотаций не важен. но лучше в проекте выдерживать единый стиль/порядок.

kataus commented on library-spring-data-mongo-mvc-view/src/main/java/ru/otus/library/service/BookServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+    public Book getBookWithCommentsById(String id) {
+        return bookRepository.findById(id)
+                .map(book -> {
+                    book.setComments(commentRepository.findAllByBookId(id));
```
Тогда зачем нам комментарии в книге?

kataus commented on library-spring-data-mongo-mvc-view/src/main/resources/templates/editBook.html:
```diff
@@ -0,0 +1,66 @@
+
+<div class="container-xxl" th:if="${book.id}">
+    <form id="edit-form" th:method="post" th:action="@{/book}" th:object="${book}">
+        <input type="text" th:field="*{id}" class="form-control" hidden="hidden"/>
```
Это единственное чем add от edit отличается, можно легко формы слить в одну

Commit:
version 1.0

1. Убраны комментарии из книги.
2. Формы add и edit сущностей объединены в одну.
3. Добавлен обработчик исключений.
